test/e2e/proxy_test.go

```diff
@@ -484,6 +487,76 @@ func TestProxyServiceSessionAffinity(t *testing.T) {
 	}
 }
 
+func testProxyExternalTrafficPolicyCase(t *testing.T, data *TestData) {
+	if len(clusterInfo.podV4NetworkCIDR) != 0 {
+		ipFamily := corev1.IPv4Protocol
+		testProxyExternalTrafficPolicy(&ipFamily, data, t)
+	}
+	if len(clusterInfo.podV6NetworkCIDR) != 0 {
+		ipFamily := corev1.IPv6Protocol
+		testProxyExternalTrafficPolicy(&ipFamily, data, t)
+	}
+}
+
+func testProxyExternalTrafficPolicy(ipFamily *corev1.IPFamily, data *TestData, t *testing.T) {
+	skipIfNumNodesLessThan(t, 2)
+	skipIfProxyDisabled(t)
+
+	svcName := fmt.Sprintf("nodeport-external-traffic-policy-test-%v", ipFamily)
+	nodes := []string{nodeName(0), nodeName(1)}
+	// Create a busybox Pod on every Node. The busybox Pod is used as a client.
+	var podIPs []*net.IP
+	for idx, node := range nodes {
+		_, ips, _ := createAndWaitForPod(t, data, data.createAgnhostPodOnNode, fmt.Sprintf("agnhost-%d-", idx), node, testNamespace, false)
+		if *ipFamily == corev1.IPv4Protocol {
+			podIPs = append(podIPs, ips.ipv4)
+		} else {
+			podIPs = append(podIPs, ips.ipv6)
+		}
+	}
+
+	// Function of checking the number of occurrences of the target IP in given OVS group output.
+	checkOutputGroup := func(groupOutput string, podIP *net.IP, expectedCount int) {
```

Comment:
Why it doesn't check the traffic and returned client ip directly but check the OVS groups? I think it's more end to end to do the former and can find issues that cannot detected by the latter check.

Reply:
Agreed. Updated.